File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        
        Map<Character,Integer> firstStringMap = new HashMap<>();
        
        for (char c: s.toCharArray()) {
            firstStringMap.put(c, firstStringMap.getOrDefault(c,0) + 1);
        }
        System.out.println(firstStringMap);
        
        for (char c : t.toCharArray()) {
            if (!firstStringMap.containsKey(c)) {
                return false;
            } 
            firstStringMap.put(c, firstStringMap.get(c) - 1);
            if (firstStringMap.get(c) == 0) {
                firstStringMap.remove(c);
            }
        }
        
        return firstStringMap.isEmpty();
    }
}

File: 0242-valid-anagram/0242-valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length !== t.length) {
        return false;
    }
    
    const firstStringMap : Map<string,number> = new Map();
    for (const char of s) {
        firstStringMap.set(char,(firstStringMap.get(char) || 0) + 1);
    }
    
    console.log(firstStringMap);
    
    for (const char of t) {
        if(!firstStringMap.has(char)) {
            return false;
        } else {
            firstStringMap.set(char,firstStringMap.get(char)! -1);
            
            if(firstStringMap.get(char) === 0) {
                firstStringMap.delete(char);
            }
        }
    }
    
    
    return firstStringMap.size === 0;
    
};